clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -112,12 +112,22 @@ PVConstraint *ConstraintResolver::addAtom(PVConstraint *PVC, ConstAtom *PtrTyp,
   return TmpPV;
 }
 
+static bool getSizeOfArg(Expr *Arg, QualType &ArgTy) {
+  if (auto *SizeOf = dyn_cast<UnaryExprOrTypeTraitExpr>(Arg))
+    if (SizeOf->getKind() == UETT_SizeOf){
+      ArgTy = SizeOf->getTypeOfArgument();
+      return true;
+    }
+  return false;
+}
+
 // Processes E from malloc(E) to discern the pointer type this will be
 static ConstAtom *analyzeAllocExpr(CallExpr *CE, Constraints &CS,
                                    QualType &ArgTy, std::string FuncName,
                                    ASTContext *Context) {
   if (FuncName.compare("calloc") == 0) {
-    ArgTy = CE->getArg(1)->getType();
+    if (!getSizeOfArg(CE->getArg(1), ArgTy))
+      return nullptr;
```

Comment:
That's interesting. If you don't pass `sizeof(...)` to `calloc` then we can't make its returned value a checked pointer? Odd that this is a problem for `calloc` but not `malloc`.

Reply:
Oh n/m; I see it's actually the same for `malloc`. The logic of this function has changed since I last looked at it.